code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm

```diff
@@ -172,22 +172,25 @@ Difficulty: Medium
 	icon = 'icons/obj/guns/magic.dmi'
 	slot_flags = ITEM_SLOT_BACK
 	w_class = WEIGHT_CLASS_BULKY
-	force = 25
+	force = 15
 	damtype = BURN
 	hitsound = 'sound/weapons/sear.ogg'
 	var/storm_type = /datum/weather/ash_storm
 	var/storm_cooldown = 0
-	var/static/list/excluded_areas = list(/area/reebe/city_of_cogs)
+	var/static/list/allowed_areas = list(/area/lavaland/surface/outdoors)
 
 /obj/item/staff/storm/attack_self(mob/user)
 	if(storm_cooldown > world.time)
 		to_chat(user, "<span class='warning'>The staff is still recharging!</span>")
 		return
 
+	if(user.z != 5)
```

Comment:
Z-Level indexes are not specifically defined, here you aren't checking if the user is on lavaland, you are checking if the user is on some arbritary z-level (that just so happens to be lavaland sometimes).

Reply:
`if(!is_mining_level(user.z))`